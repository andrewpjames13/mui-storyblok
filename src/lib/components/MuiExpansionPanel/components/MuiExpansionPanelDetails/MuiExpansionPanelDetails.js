import React, { createElement } from 'react';
import PropTypes from 'prop-types';
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
import Storyblok from '../../../../utils/Storyblok';
import MuiTypography from '../../../MuiTypography/MuiTypography';

const MuiExpansionPanelDetails = ({ rootClass, content }) => {
  const components = {
    MuiTypography,
  };

  const styles = Storyblok.arrayToMuiStyles(rootClass);

  return (
    <ExpansionPanelDetails
      className={styles.root}
    >
      {content.map((item, index) => createElement(
        components[item.component],
        Object.assign(item, { key: index }),
      ))}
    </ExpansionPanelDetails>
  );
};

export default MuiExpansionPanelDetails;

MuiExpansionPanelDetails.propTypes = {
  /** stroyblok multiselect of css classes */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /** MuiTypography */
  content: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,
};

MuiExpansionPanelDetails.defaultProps = {
  rootClass: [],
};
