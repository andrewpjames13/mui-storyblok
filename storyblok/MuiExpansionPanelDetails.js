module.exports = {
  "name": "MuiExpansionPanelDetails",
  "display_name": null,
  "updated_at": "2020-06-11T21:21:48.600Z",
  "schema": {
    "rootClass": {
      "type": "options"
    },
    "content": {
      "type": "bloks",
      "restrict_components": true,
      "component_whitelist": [
        "MuiTypography"
      ]
    }
  },
  "image": null,
  "preview_field": null,
  "is_root": false,
  "preview_tmpl": null,
  "is_nestable": true,
  "all_presets": [],
  "preset_id": null,
  "real_name": "MuiExpansionPanelDetails",
  "component_group_uuid": null
}