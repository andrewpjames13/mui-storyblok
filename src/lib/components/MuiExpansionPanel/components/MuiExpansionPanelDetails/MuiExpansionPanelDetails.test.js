import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import MuiExpansionPanelDetails from './MuiExpansionPanelDetails';

function setup() {
  const props = {
    content: [{
      component: 'MuiExpansionPanelTypography',
      content: [{
        component: 'MuiText',
        text: 'text',
      }],
    }],
  };
  const comp = shallow(<MuiExpansionPanelDetails {...props} />);
  return { comp, props };
}

describe('<MuiExpansionPanelDetails />', () => {
  it('renders MuiExpansionPanelDetails', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();

    const tree = renderer.create(<MuiExpansionPanelDetails {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
