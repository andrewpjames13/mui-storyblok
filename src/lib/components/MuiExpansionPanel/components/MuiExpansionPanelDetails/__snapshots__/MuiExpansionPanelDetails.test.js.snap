// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiExpansionPanelDetails /> snapshot 1`] = `
<div
  className="MuiExpansionPanelDetails-root makeStyles-root-3"
>
  <div
    className="MuiTypography-root makeStyles-root-5 MuiTypography-body1"
  >
    <span
      className="MuiTypography-root makeStyles-root-6 MuiTypography-body1"
    >
       text 
    </span>
  </div>
</div>
`;
